### PR TITLE
Filter all files that live outside of `#{CREW_PREFIX}`, put them in `#{CREW_PREFIX}/${orig_path}` instead

### LINKED ISSUE #7081: Unable to install dbus
```
$ crew install dbus
dbus: D-Bus is a message bus system, a simple way for applications to talk to one another.
https://www.freedesktop.org/wiki/Software/dbus/
Version: 1.13.18-1
License: Apache-2.0
Performing pre-flight checks...
Precompiled binary available, downloading...
[##############################################################################################################################################################################################]   2.10 MB 100%
Dbus archive downloaded.
Unpacking 'tpxz' archive using 'tar', this may take a while...
Performing pre-install...
Performing install...
Using rdfind to convert duplicate files to hard links.
Now scanning ".", found 50 files.
Now have 50 files in total.
Removed 0 files due to nonunique device and inode.
Total size is 3514013 bytes or 3 MiB
Removed 48 files due to unique sizes from list. 2 files left.
Now eliminating candidates based on first bytes: removed 2 files from list. 0 files left.
Now eliminating candidates based on last bytes: removed 0 files from list. 0 files left.
Now eliminating candidates based on sha1 checksum: removed 0 files from list. 0 files left.
It seems like you have 0 files that are not unique
Totally, 0 B can be reduced.
Now making hard links.
Making 0 links.
/usr/local/lib/ruby/3.1.0/fileutils.rb:243:in `mkdir': Permission denied @ dir_s_mkdir - /var/run (Errno::EACCES)
	from /usr/local/lib/ruby/3.1.0/fileutils.rb:243:in `fu_mkdir'
	from /usr/local/lib/ruby/3.1.0/fileutils.rb:221:in `block (2 levels) in mkdir_p'
	from /usr/local/lib/ruby/3.1.0/fileutils.rb:219:in `reverse_each'
	from /usr/local/lib/ruby/3.1.0/fileutils.rb:219:in `block in mkdir_p'
	from /usr/local/lib/ruby/3.1.0/fileutils.rb:211:in `each'
	from /usr/local/lib/ruby/3.1.0/fileutils.rb:211:in `mkdir_p'
	from /usr/local/bin/crew:1385:in `block in install_package'
	from /usr/local/bin/crew:1331:in `chdir'
	from /usr/local/bin/crew:1331:in `install_package'
	from /usr/local/bin/crew:1541:in `install'
	from /usr/local/bin/crew:1406:in `resolve_dependencies_and_install'
	from /usr/local/bin/crew:1940:in `block in install_command'
	from /usr/local/bin/crew:1934:in `each'
	from /usr/local/bin/crew:1934:in `install_command'
	from /usr/local/bin/crew:2060:in `<main>'
```

### PR DESCRIPTION
**Pull request by supechicken**
Fixes #7081 
Revert #7090 

The reason that the new install breaks on some packages is some packages have files outside of `#{CREW_PREFIX}`, and `crew` will try to `mkdir` on those places, which cause errors like `permission denied` and `read-only filesystem`.

This PR adds a filter for `directoryList` and `filelist` that will redirect all paths outside `#{CREW_PREFIX}` to `#{CREW_PREFIX}/#{their_original_path}`, also added a warning for that.

(btw sorry for ignoring the issue 😅)

Changes are in around line 1348

Tested on `x86_64`

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING=1 CREW_TESTING_REPO=https://github.com/skycocker/chromebrew CREW_TESTING_BRANCH=revert-7090-revert-d3038e08050cb3170b0414446f1cb85411aa4c9f crew update
crew reinstall dbus::all
```

### CODE REVIEW

File: packages/libxml2.rb [hunk 1]
```diff
@@ -22,7 +22,6 @@ class Libxml2 < Package
     x86_64: 'ebb3b81155231be702617246a78945a3698142f63b3138c5763b9813d57dbe19'
   })
 
-  depends_on 'gcc'
```
Comment (uberhacker) [inline packages/libxml2.rb:25]: Add `depends_on 'gcc::libs'`?

**Comment by uberhacker** [comment]
Bump crew version in `lib/const.rb`?

**Comment by saltedcoffii** [comment]
Many packages do use `/var` and that seems acceptable. Will this accommodate that?

**Comment by uberhacker** [comment]
The `/var` directory is not writable.  All package files have to go into a directory accessible by the `chronos` user.  By default, this is `/usr/local`.

File: packages/libxml2.rb (hunk 1 above)
Comment (supechicken) [reply packages/libxml2.rb:25]: This is a deleted line (`libxml2` does not depend on `gcc`)

**Comment by supechicken** [comment]
> Bump crew version in `lib/const.rb`?

Done

**Pull request closed without merging**